src/prototype_creep_routing.js

```diff
@@ -162,42 +162,62 @@ Creep.prototype.getDirections = function(path, pathPos) {
   };
 };
 
+Creep.prototype.checkPreMove = function(datas) {
+  if (!datas.path) {
+    return true;
+  }
+  if (datas.pathPos < 0) {
+    this.memory.routing.pathPos = datas.pathPos;
+    return true;
+  }
+  return false;
+};
+
+Creep.prototype.checkContinue = function(datas) {
+  if (datas.path && datas.pathPos < 0 && datas.path.length === 0) {
+    return true;
+  }
+  return false;
+};
+
 Creep.prototype.followPath = function(action) {
   const route = this.getRoute();
   const routePos = this.getRoutePos(route);
 
   // TODO Disable base room for now
   // if (routePos === 0) {
-  // this.say('R:Base');
-  // return false;
+  //   this.say('R:Base');
+  //   return false;
   // }
 
   if (!this.memory.routing.targetId && this.room.name === this.memory.routing.targetRoom) {
     this.memory.routing.reached = true;
     return action(this);
   }
   const prepareData = this.moveByPathPrepare(route, routePos, 'pathStart', this.memory.routing.targetId);
-  if (prepareData.unit.preMove) {
-    if (prepareData.unit.preMove(this, prepareData.directions)) {
+  if (this.checkPreMove(prepareData)) {
+    if (prepareData.unit.preMove && prepareData.unit.preMove(this, prepareData.directions)) {
       return true;
     }
+    if (!this.checkContinue(prepareData)) {
```

Comment:
Do we really need to extract `checkContinue`? I think it would be easier to understand it inline